std/file.d

```diff
@@ -4444,6 +4445,7 @@ void rmdirRecurse(scope const(char)[] pathname)
 }
 
 /// ditto
+@trusted
```

Comment:
What needs to be `@trusted` in this function?

Reply:
`dirEntries` is @system, so that's why it needs @trusted attribute. In this context, I consider this being safe because `dirEntries` returns a RefCounted wrapped variable, that, inside this will always be eliminated, so it doesn't make much sense to be unsafe here, IMHO.

The unsafe part is here:
```d
foreach (DirEntry e; dirEntries(de.name, SpanMode.depth, false))
{
    attrIsDir(e.linkAttributes) ? rmdir(e.name) : remove(e.name);
}
```

Reply:
is it the call to dirEntries, or the iteration or both? What about `rmdir`/`remove`?

Reply:
Can't we use a local @trusted lambda + a comment to make sure nothing unsafe gets added in the future?

Reply:
> is it the call to dirEntries, or the iteration or both? What about `rmdir`/`remove`?

both `rmdir` and `remove` uses a @trusted `trustedRmdir` and `removeImpl` respectively, so, when called it's both @safe calls. 

> Can't we use a local @trusted lambda + a comment to make sure nothing unsafe gets added in the future?

Sure, I'll do it.

Also, if you follow this example here, https://github.com/dlang/phobos/blob/453faadf5bc25d2f4f4ecd81133f1a9317564bde/std/datetime/timezone.d#L2451 , they are also using `dirEntries` in a @trusted function, so maybe it should be a good idea to wrap it too. I'll do it in another PR.